ifsbench/arch.py

```diff
@@ -91,54 +94,53 @@ def process_job(self, job: Job, **kwargs):
         """
 
 
-class DefaultArch(Arch):
+class DefaultArch(Arch, PydanticConfigMixin):
 
-    def __init__(
-        self,
-        launcher: Launcher,
-        cpu_config: CpuConfiguration,
-        set_explicit: bool = False,
-        account: Optional[str] = None,
-        partition: Optional[str] = None,
-    ):
-        """
-        Default architecture that can be used for various systems.
+    """
+    Default architecture that can be used for various systems.
+
+    Parameters
+    ----------
+    launcher : Launcher
```

Comment:
As you've taken away the constructor, could you move the parts of the docstring that describe the parameters directly to the variable definitions (example below). If I remember, this is necessary for Sphinx being able to pick this up correctly.
```suggestion
    #: The default launcher that is used on this system.
    launcher : Launcher
    
    #: The hardware setup of the system.
    cpu_config: CpuConfiguration
    
    #: If set to True, the following attributes in result.job are
    #: calculated and set explicitly, using ``cpu_config``:
    #:      * tasks
    #:      * nodes
    #:      * tasks_per_node
    #:  If not, these values will stay None, if not specified.
    set_explicit: bool = False
    
    ...
```